### DISCUSSION TITLE
Debugger and Step-by-Step Debugging Tools

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [zed-industries/zed#6566] [Feature Requests]
There are plans to include debugging tools in the upcoming features or future roadmap for Zed? Specifically, it would be invaluable to have features such as:

- **Integrated Debugger**: A built-in debugger that could allow us to inspect our code within the Zed environment, without needing to switch to a different tool or environment.
- **Breakpoints**: The ability to set breakpoints in the code, which would pause the execution of the program at those points. This would be very useful for checking the state of variables and understanding the flow of the program.
- **Step-by-step Debugging**: This feature would allow us to execute our code line by line, making it easier to spot and understand where and why errors occur.

**Comment by iamnbutler** [comment]
I don't think we have any immediate plans for this, but it is highly requested.

Relevant:

- https://github.com/zed-industries/community/issues/811
- https://github.com/zed-industries/community/issues/668
- https://github.com/zed-industries/community/issues/455
- https://github.com/zed-industries/community/issues/267

**Comment by ScopeSV** [comment]
Nice. This is kind of a deal breaker if you want engineers to migrate 100% over to zed. 
Cant change editors each time I need to debug something :)

**Reply by semiautomatix** [reply]
Yeah, I've had to go back to Cursor to debug the issue. This is a serious deficit.

**Reply by iamnbutler** [reply]
Ultimately, some developers need a debugger and can't live without one, while others never use one. 

For now, Zed won't work for the former. 

We'll get to it, we know people want it–but we'll take the time to do it right over adding something quickly that could introduce both design and engineering technical debt.

It sucks when that is the one thing keeping you from using Zed, I understand that! Hopefully it comes quicker than you expect :)

**Reply by smasyutin** [reply]
@iamnbutler add my vote for debugger support. Unfortunately, "no debugger" is a "no-go" for Python backend development.
